AmplifyTests/CategoryTests/Auth/AuthCategoryConfigurationTests.swift

```diff
@@ -0,0 +1,367 @@
+                "MockSecondAuthCategoryPlugin": true
+            ]
+        )
+
+        let amplifyConfig = AmplifyConfiguration(auth: config)
+
+        try Amplify.configure(amplifyConfig)
+
+        let exception: BadInstructionException? = catchBadInstruction {
+            _ = Amplify.Auth.changePassword(currentPassword: "current",
+                                            newPassword: "new",
+                                            listener: nil)
+        }
+        XCTAssertNotNil(exception)
+    }
+
+    /// Test if configuration Auth plugin directly works
+    ///
+    /// - Given: Amplify with Auth plugin configured
+    /// - When:
+    ///    - I try to add a new configuration to the same plugin
+    /// - Then:
+    ///    - Should work without any error.
+    ///
+    func testCanConfigurePluginDirectly() throws {
+        let plugin = MockAuthCategoryPlugin()
+        let configureShouldBeInvokedFromCategory =
+            expectation(description: "Configure should be invoked by Amplify.configure()")
+        let configureShouldBeInvokedDirectly =
+            expectation(description: "Configure should be invoked by getPlugin().configure()")
+
+        var invocationCount = 0
+        plugin.listeners.append { message in
+            if message == "configure(using:)" {
+                invocationCount += 1
+                switch invocationCount {
+                case 1: configureShouldBeInvokedFromCategory.fulfill()
+                case 2: configureShouldBeInvokedDirectly.fulfill()
+                default: XCTFail("Expected configure() to be called only two times, but got \(invocationCount)")
+                }
+            }
+        }
+        try Amplify.add(plugin: plugin)
+
+        let config = AuthCategoryConfiguration(
+            plugins: ["MockAuthCategoryPlugin": true]
+        )
+
+        let amplifyConfig = AmplifyConfiguration(auth: config)
+
+        try Amplify.configure(amplifyConfig)
+        try Amplify.Auth.getPlugin(for: "MockAuthCategoryPlugin").configure(using: true)
+        waitForExpectations(timeout: 1.0)
+    }
+
+    /// Test if unconfigured plugin throws an error
+    ///
+    /// - Given: An unconfigured Amplify framework with Auth plugin added
+    /// - When:
+    ///    - Invoke an API in auth
+    /// - Then:
+    ///    - I should get an exception
+    ///
+    func testPreconditionFailureInvokingBeforeConfig() throws {
+        let plugin = MockAuthCategoryPlugin()
+        try Amplify.add(plugin: plugin)
+
+        // Remember, this test must be invoked with a category that doesn't include an Amplify-supplied default plugin
+        let exception: BadInstructionException? = catchBadInstruction {
+            _ = Amplify.Auth.changePassword(currentPassword: "current",
+                                            newPassword: "new",
+                                            listener: nil)
+        }
+        XCTAssertNotNil(exception)
```

Comment:
maybe check for the not configured exception rather than just not nil?

Reply:
This is a preconditionfailure, I am not sure we can check the type here.